### PR TITLE
Adds new functionality for equiprobable bin plotting

### PR DESCRIPTION
**Pull request by Isaac-Sarnoff**
This modification allows for the ability to use the equiprobable bin plotting methods while showing either the actual number of counts in each bin , or the density of those counts through the use of the argument plot_mode, which can be set to 'num_counts' and 'count_density' respectively. plot_mode will still default to "sigma_deviation", which plots the sigma deviation of the number of counts in each bin from the expectation. In order to make this change, there have been modifications to bin_equiprobable in evaluator_base, and to plot_equiprobable_histogram in utils. Also in utils, two new helper functions have been added. The first, limit_bin_edges sets non-infinite outer edges on histogram bins so that their area can be calculated. The second, density_modifier, scales the number of hits in each bin by the area of that bin. This is used for 'count_density' mode. 

Below is some adapted tutorial code that can be used to demonstrate the use of these new functions:
```python
import GOFevaluation as ge
import scipy.stats as sps
import matplotlib.pyplot as plt
import numpy as np

# Generate toy data sample and reference sample
reference_sample = sps.norm().rvs([125*100, 2], random_state=42)
reference_sample.T[1] += reference_sample.T[0]

data_sample = sps.norm().rvs([125, 2], random_state=43)
data_sample.T[1] += data_sample.T[0]

plt.scatter(reference_sample.T[0], reference_sample.T[1])
plt.scatter(data_sample.T[0], data_sample.T[1])
plt.show()
```
image

This creates a set of both reference distribution points and a set of data points that will be plotted according to the equiprobable binning technique.

```python
# define eqpb parameters
order = [0, 1]
n_partitions = [8, 8]

unusedterm, bin_edges = ge.equiprobable_histogram(
            data_sample=reference_sample,
            reference_sample=reference_sample,
            n_partitions=n_partitions,
            order=order)

ge.plot_equiprobable_histogram(data_sample=data_sample,
                                        bin_edges=bin_edges,
                                        order=order,
                                        nevents_expected=125,
                                        )
ax = plt.gca()
ax.set_xlabel('$x_1$')
ax.set_ylabel('$x_2$')
```
image

This shows the standard method of using the equiprobable binning plotting with sigma deviations. Notice that we need to define some of our terms before hand. One way to get around this, shown in the next code sample creates a BinnedChi2GOF object, a subclass of evaluator base. This can do some of the setup. Which particular subclass you use will change the GOF computations, but will not change the plotting, and we can therefore do it freely.

```python
ge.BinnedChi2GOF.bin_equiprobable(data_sample=data_sample,
                                                      reference_sample=reference_sample,
                                                      nevents_expected=125,
                                                      n_partitions=n_partitions,
                                                      order=order,
                                                      plot=True,
                                                      plot_mode='num_counts',
                                                     )
ax = plt.gca()
ax.set_xlabel('$x_1$')
ax.set_ylabel('$x_2$')
```
 
image

In the above code, in addition to using the object instantiation mentioned above, we also the 'num_counts' plot_mode argument.

Finally, we use the 'count_density' argument, as well as setting cmap='jet' just for fun.

```python
ge.BinnedPoissonChi2GOF.bin_equiprobable(data_sample=data_sample,
                                                      reference_sample=reference_sample,
                                                      nevents_expected=125,
                                                      n_partitions=n_partitions,
                                                      order=order,
                                                      plot=True,
                                                      cmap='jet',
                                                      plot_mode='count_density'
                                                     )
ax = plt.gca()
ax.set_xlabel('$x_1$')
ax.set_ylabel('$x_2$')
```
image

### CODE REVIEW

**Review by hammannr — COMMENTED** [review]
This looks really nice! Thanks for making this new clean PR!! 😊 
I made a few comments that I missed in the earlier stage. Also, please add ``.DS_Store`` to the ``gitignore`` file so that we don't include it in the PR.
Apart from that, there are a bunch of flake8 issues. The easiest way to check [github Actions](https://github.com/XENONnT/GOFevaluation/runs/4950791853?check_suite_focus=true). Soe things are straightforward I think but if anything is unclear I can have a look how to solve it. Since it failed the flake8 checks, the actions didn't proceed with unittests. But if I run pytest locally, it will raise an error while importing ``_get_finite_bin_edges``. Not sure what's going on there tbh... Does it run through on your local version?
If you make a new version with the few adjustments I will run the code and if everything looks good we can merge! 👌🏼

File: GOFevaluation/evaluator_base.py [hunk 1]
```diff
@@ -109,10 +109,15 @@ def bin_equiprobable(cls, data_sample, reference_sample, nevents_expected,
...
         if plot:
+            try:
+                plot_mode = kwargs.pop('plot_mode')
+            except KeyError:
+                pass
             plot_equiprobable_histogram(data_sample=data_sample,
                                         bin_edges=bin_edges,
                                         order=order,
                                         nevents_expected=nevents_expected,
+                                        plot_mode=plot_mode,
```
[the remark this answers is not in the document]
Comment (WolfXeHD) [reply GOFevaluation/evaluator_base.py:120]: Without having checked this in detail: Do we really need a try&except statement there? Can't we pull a default argument if 'plot_mode' is not provided?

[the remark this answers is not in the document]
Comment (hammannr) [reply GOFevaluation/evaluator_base.py:120]: yes of course you're right.. much easier!

**Review by hammannr — COMMENTED** [review]
looks good so far. I'll fix the whitespace and too-long line errors, go through the new tests and run the code. If everything works well I think the PR is good to go! 🚀

**Review by hammannr — APPROVED** [review]
I fixed a few more flake8 issues and simplified `test__get_count_density`. Apart from that, I introduced absolute imports for all methods of GOFevaluation (I got an error in `test_utils` due to this and I think it's best practice anyways) and changed the label of the colorbar for `sigma_deviation` to be a bit more meaningful (it now also shows the expectation in each bin).

@Isaac-Sarnoff , if you're happy with the minor adjustments I made, you can go ahead and merge your PR! Many thanks for this very nice contribution to the package!! 🙌

**Pull request merged**